Update documentation on user storage provider in Quarkus (#19898)
keycloak/keycloak: docs/documentation/server_development/topics/user-storage/registration-query.adoc

    }

    @Override
    public Stream<UserModel> searchForUserStream(RealmModel realm, String search, Integer firstResult, Integer maxResults) {
        Predicate<String> predicate = "*".equals(search) ? username -> true : username -> username.contains(search);
        return properties.keySet().stream()
                .map(String.class::cast)
                .filter(predicate)
                .skip(firstResult)
                .map(username -> getUserByUsername(realm, username))
                .limit(maxResults);
    }
----

> The first declaration of `searchForUserStream()` takes a `String` parameter. In this example, it represents a username that you want to search by. This string can be a substring, which is why we use the `String.contains()`
> method when doing our search. Notice that we use `*` to indicate we want to list all users.
> The method iterates over the key set of the property file, delegating to `getUserByUsername()` to load a user.

Reviewer: ```suggestion
The first declaration of `searchForUserStream()` takes a `String` parameter. In this example, the parameter represents a username that you want to search by. This string can be a substring, which explains the choice of the `String.contains()`
method when doing the search. Notice the use of `*` to indicate to request a list all users.
The method iterates over the key set of the property file, delegating to `getUserByUsername()` to load a user.
```

Notice that we are indexing this call based on the `firstResult` and `maxResults` parameter. If your external store does not support pagination, you will have to do similar logic.

.PropertyFileUserStorageProvider
[source,java]
----
    @Override
    public Stream<UserModel> searchForUserStream(RealmModel realm, Map<String, String> params, Integer firstResult, Integer maxResults) {
        // only support searching by username
        String usernameSearchString = params.get("username");
        if (usernameSearchString != null)
            return searchForUserStream(realm, usernameSearchString, firstResult, maxResults);

        // if we are not searching by username, return all users
        return searchForUserStream(realm, "*", firstResult, maxResults);
    }
----

The `searchForUserStream()` method that takes a `Map` parameter can search for a user based on first, last name, username, and email.

> We only store usernames, so we only search based on usernames. Only exception is when `Map` parameter doesn't contain `username` attribute
> in which case we return all users. We delegate to `searchForUserStream(realm, search, firstResult, maxResults)` for this.

Reviewer: ```suggestion
Only usernames are stored, so the search is based only on usernames except when the `Map` parameter does not contain the `username` attribute. In this case, all users are returned. In that situation,  use `searchForUserStream(realm, search, firstResult, maxResults)` .
```

.PropertyFileUserStorageProvider
[source,java]
----
    @Override
    public Stream<UserModel> getGroupMembersStream(RealmModel realm, GroupModel group, Integer firstResult, Integer maxResults) {
        return Stream.empty();
    }

    @Override
    public Stream<UserModel> searchForUserByUserAttributeStream(RealmModel realm, String attrName, String attrValue) {
        return Stream.empty();
    }
----

> We do not store groups or attributes, so the other methods return an empty stream.

Reviewer: ```suggestion
Groups or attributes are not stored, so the other methods return an empty stream.
```
